### DISCUSSION TITLE
battery usage

### DISCUSSION THREAD

**Discussion by GabriF** [flxzt/rnote#817] [General]
Hello, I'm using rnote to take notes in class with my stylus. However, it seems like it dries the battery a lot. Like a 10 minutes usage is 5%, which is -30%/hour. My laptop's fans are also fast during the usage and it becames hot. Is it normal? Am I the only one facing this problem? I don't have this problem with other apps and similar as xournalpp. CPU usage is low, it takes just 1GB of ram.

This is my system:
Laptop model: Lenovo IdeaPad Flex 5 14inch
CPU: Amd Ryzen 5 7530U
Ram: 16GB

OS: Arch Linux
Kernel version: 6.5.5
Rnote version: 0.8.2-1

**Comment by Haislich** [comment]
Can confirm that battery usage is quite high in windows too.
The tablet gets hot within minutes and battery consumption goes through the roof.
I've also noted that when pdf are imported the system lags.

Laptop model: Surface pro 8
CPU: Intel  i7-1185G7
Ram: 16GB

OS: Windows
Rnote version: 0.8.2

**Comment by Doublonmousse** [comment]
From what I've seen, the GPU usage is quite high in rnote (on surface pro 8, windows), and it seems to be higher than some other note-taking app on the windows side, although the ones I compared rnote with also had non-trivial GPU utilisation.

If you look at GPU usage while drawing, you see a spike in GPU usage.

**Reply by GabriF** [reply]
I tried to write with stylus with htop near and CPU was between 95-98% all the time

**Reply by Doublonmousse** [reply]
So I tried and I get some CPU and GPU usage when writing. But nowhere near 95-98% CPU. 
My guess would be that the GPU acceleration is not working and hardware acceleration is used instead, putting all the work on the CPU instead of CPU + GPU

**Reply by GabriF** [reply]
Are you using wayland or xorg? I was using it on wayland and it seems to use less CPU on x (xwayland). If you are on wayland you can  force it to use xwayland launching it this way ```WAYLAND_DISPLAY="" rnote```

**Reply by Doublonmousse** [reply]
I'm using it on Windows. So there may be some windows+gtk4 effect that would make it behave differently than on linux.
I have not tested it on linux for now.

**Reply by Doublonmousse** [reply]
And I'm compiling from source on windows...

**Reply by Doublonmousse** [reply]
On linux, I've tested different things, changing the flatpak parameters in flatseal:
- gpu acceleration (on or off)
- Wayland (on or off)

using Fedora 38 and gnome 44.5 with Wayland enabled

What I've found
- With gpu acceleration enabled, there seems to be slightly higher GPU usage with Wayland than X, but not much difference on the CPU utilization (at MOST 30 % on a single thread when writing or immediately after writing, 30 % being the CPU % utilization unit seen by htop)
- With gpu acceleration disabled, the CPU usage is very high both with or without Wayland, and slightly lower with X. This time, all cores are used (I can get all cores up to 60-70 % on my machine), and the GPU is doing very little
